build.sbt

```diff
@@ -44,6 +53,7 @@ def genericModule(proj: Project) = proj
   .dependsOn(testkit % "test")
   .settings(commonSettings)
   .settings(name := s"pureconfig-${baseDirectory.value.getName}")
+  .settings(Seq(javaModuleName := s"${name.value.replace("-", ".")}") ++ publishSettingAutomaticModuleName)
```

Comment:
I don't understand why we're defining this only for `core` and generic modules. Don't other modules, like `pekko-http` and `scala-xml`, suffer from the same problem? If we can, we should skip the SBT setting boilerplate and just inline this into `commonSettings`.